analyzers/src/SonarAnalyzer.CSharp/Rules/RedundantDeclaration.cs

```diff
@@ -291,66 +268,56 @@ private static void ReportOnRedundantObjectInitializer(SyntaxNodeAnalysisContext
 
         private static void ReportOnExplicitDelegateCreation(SyntaxNodeAnalysisContext context)
         {
-            var objectCreation = (ObjectCreationExpressionSyntax)context.Node;
-            if (!IsDelegateCreation(objectCreation, context.SemanticModel))
+            var objectCreation = ObjectCreationFactory.Create(context.Node);
+            var argumentExpression = objectCreation.ArgumentList?.Arguments.FirstOrDefault()?.Expression;
+            if (argumentExpression == null)
             {
                 return;
             }
 
-            var argumentExpression = objectCreation.ArgumentList?.Arguments.FirstOrDefault()?.Expression;
-            if (argumentExpression == null)
+            if (!IsDelegateCreation(objectCreation, context.SemanticModel))
             {
                 return;
             }
 
-            if (IsInDeclarationNotVarNotDelegate(objectCreation, context.SemanticModel) ||
-                IsAssignmentNotDelegate(objectCreation, context.SemanticModel) ||
-                IsReturnValueNotDelegate(objectCreation, context.SemanticModel) ||
-                IsInArgumentAndCanBeChanged(objectCreation, context.SemanticModel,
+            if (IsInDeclarationNotVarNotDelegate(objectCreation.Expression, context.SemanticModel)
+                || IsAssignmentNotDelegate(objectCreation.Expression, context.SemanticModel)
+                || IsReturnValueNotDelegate(objectCreation.Expression, context.SemanticModel)
+                || IsInArgumentAndCanBeChanged(objectCreation, context.SemanticModel,
                     invocation => invocation.ArgumentList.Arguments.Any(a => IsDynamic(a, context.SemanticModel))))
             {
-                ReportIssueOnRedundantObjectCreation(context, objectCreation, "explicit delegate creation", RedundancyType.ExplicitDelegate);
+                ReportIssueOnRedundantObjectCreation(context, objectCreation.Expression, "explicit delegate creation", RedundancyType.ExplicitDelegate);
             }
         }
 
-        private static bool IsDynamic(ArgumentSyntax argument, SemanticModel semanticModel)
-        {
-            return semanticModel.GetTypeInfo(argument.Expression).Type is IDynamicTypeSymbol;
-        }
+        private static bool IsDynamic(ArgumentSyntax argument, SemanticModel semanticModel) =>
+            semanticModel.GetTypeInfo(argument.Expression).Type is IDynamicTypeSymbol;
 
-        private static bool IsInDeclarationNotVarNotDelegate(ObjectCreationExpressionSyntax objectCreation, SemanticModel semanticModel)
+        private static bool IsInDeclarationNotVarNotDelegate(SyntaxNode objectCreation, SemanticModel semanticModel)
```

Comment:
Similar as the previous: Let's add a C# 9 compliant test case that will enter this section and find the variable declarations. Just to be sure it works as expected

Reply:
This one doesn't hit the breakpoint for the new `new()` syntax, only for the old one - probably related to the comment in C# 9 test case.

Reply:
Verified that we stop on this case now with implicit syntax.